### PR TITLE
Add support for webpack2 #50

### PR DESCRIPTION
**Pull request by nstojiljkovic**
Please review :)

### CODE REVIEW

**Comment by julienrf** [comment]
test this please

**Comment by julienrf** [comment]
> Checking out Revision f8fad4828b0c476c1c7fd488dd0ef2fcd727add7 (origin/yarn-appveyor)

I don’t understand why has Jenkins checked out the wrong branch…

**Comment by julienrf** [comment]
test this please

**Comment by julienrf** [comment]
test this please

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/NpmPackage.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  def major: Option[Int] = {
+    val r = """^(\d+)(\..*|)$""".r
+    r findFirstMatchIn version flatMap { case r(v, x) =>
+      try {
+        Some(v.toInt)
+      } catch {
+        case e: Exception => None
+      }
+    }
+  }
```
Comment (julienrf) [inline sbt-scalajs-bundler/src/main/scala/scalajsbundler/NpmPackage.scala]: I think you could rewrite this as follows:

~~~ scala
version match {
  case r(v, _) => Try(v.toInt).toOption
  case _ => None
}
~~~

File: sbt-scalajs-bundler/src/main/scala/scalajsbundler/NpmPackage.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  def getForModule(targetDir: File, module: String): Option[NpmPackage] = {
+    val webpackPackageJsonFilePath = targetDir / "node_modules" / module / "package.json"
+
+    try {
+      Some(json.fromJSON[NpmPackage](json.readJSON(new FileReader(webpackPackageJsonFilePath))))
+    } catch {
+      case e : Exception => None
+    }
```
Comment (julienrf) [inline sbt-scalajs-bundler/src/main/scala/scalajsbundler/NpmPackage.scala]: I prefer `Try(…).toOption`

**Comment by nstojiljkovic** [comment]
We've again got the error: "Error: Cannot find module 'C:\Users\appveyor\AppData\Local\Temp\1\sbt_492497ae\static2\target\scala-2.11\node_modules\webpack\bin\webpack'". @julienrf how did you fix it the first time?

**Comment by julienrf** [comment]
I didn’t fix this issue yet :(

**Comment by julienrf** [comment]
test this please

**Comment by julienrf** [comment]
The PR is great but, I don’t know why, there is an issue in AppVeyor: some files are not correctly removed by the `sbt clean` tasks… I will put the PR in stand by as long as I can not resolve this problem.

**Comment by julienrf** [comment]
So, I’m going to merge the PR to an internal `webpack2` branch instead of `master`.

**Pull request merged**